Resolves #2976: Add missing crosswalk tag

yutong23 opened the pull request:
Resolves #2976 
- Added a tag named "missing crosswalk" in Crosswalk label

##### Before/After screen
![image](https://user-images.githubusercontent.com/78244145/187122364-1ccc349b-52e2-44b8-9d07-4bab8478a8f9.png)
![217041661740963_ pic](https://user-images.githubusercontent.com/78244145/187121676-b69ae394-77f5-4832-9865-e6f1134f2820.jpg)
shots (if applicable)

##### Testing instructions
1. Click the Crosswalk label and pick a location to label this label
2. Click the missing crosswalk tag or use shift+J(J is the KeyChar)

misaugstad commented:
Thanks for trying to get this done quickly! There are some issues here to deal with though:
1. It looks like you branched off of your `2961-add-label-button-translation` instead of branching off of develop, so you have your code from that PR wrapped up in here. Can you try starting over, branching off of develop instead?
2. For keyboard shortcuts, in general we want to have the shortcuts match a letter in the text for the tag. And then the underscore in the text indicates which character can be used as the keyboard shortcut for the tag! So ideally we won't use 'J', since there is no J in 'missing crosswalk'. I believe that 'A' won't clash, so let's go with that for now.
4. Can you add translations for Spanish and Dutch as well? Let's just use Google Translate for now, and I'll get more permanent translations in the future!
5. And finally, although your changes to `UtilitiesSidewalk.js` mirror our other PRs that add tags, the rest of your code does not. Can you take another look at those old PRs and try to mirror them? And feel free to messages the interns channel on Slack if you are having a hard time figuring out what's going on in those old PRs!

yutong23 commented:
1. I added a new branch 2976-add-missing-crosswalk-tag from develop branch and push the code to this branch. But I found that the changes of `public/javascripts/SVLabel/src/SVLabel/canvas/ContextMenu.js` and `public/javascripts/SVValidate/src/label/LabelVisibilityControl.js` still exist in `Files Changes`. I am not sure how to delete them.
2. Solved it.
3. Solved it.
4. I added a new SQL file representing 'missing crosswalk' in the Crosswalk label.

misaugstad commented on public/javascripts/common/UtilitiesSidewalk.js:
```diff
@@ -324,6 +324,10 @@ function UtilitiesMisc (JSON) {
+                    },
+                    'missing crosswalk': {
+                        keyChar: 'A',
+                        text: i18next.t('center-ui.context-menu.tag.missing-crosswalk')
```
You're referencing a translation here, but you haven't added translations yet!

misaugstad reviewed (changes requested):
>I added a new branch 2976-add-missing-crosswalk-tag from develop branch and push the code to this branch. But I found that the changes of public/javascripts/SVLabel/src/SVLabel/canvas/ContextMenu.js and public/javascripts/SVValidate/src/label/LabelVisibilityControl.js still exist in Files Changes. I am not sure how to delete them.

I googled around real quick and found [this](https://stackoverflow.com/questions/2938301/remove-specific-commit)! Ultimately I used `git revert --strategy resolve <commit>` to remove the commit that we didn't want! So this has been fixed.

And now it almost looks good, you're just missing translations! The PR I pointed you to last time includes translations. You can just use Google Translate for now!

misaugstad commented on public/locales/en/audit.json:
```diff
@@ -169,8 +169,8 @@
                 "APS": "<tag-underline>A</tag-underline>PS",
-                "missing-crosswalk": "m<tag-underline>i</tag-underline>ssing crosswalk",
-                "no-bus-stop-access": "no bus stop <tag-underline>a</tag-underline>ccess",
+                "missing-crosswalk": "missing crosswalk (<tag-underline>a</tag-underline>)",
```
you can underline the letter 'a' in 'missing crosswalk' instead of having it in parens at the end!

misaugstad commented on public/locales/en/audit.json:
```diff
@@ -169,8 +169,8 @@
-                "missing-crosswalk": "m<tag-underline>i</tag-underline>ssing crosswalk",
-                "no-bus-stop-access": "no bus stop <tag-underline>a</tag-underline>ccess",
+                "missing-crosswalk": "missing crosswalk (<tag-underline>a</tag-underline>)",
+                "no-bus-stop-access": "no bus stop access (<tag-underline>u</tag-underline>)",
```
same with the 'u' in 'no bus stop access'

misaugstad reviewed (changes requested):
I think we are almost there! You just want to underline the letter in the word that can be used as the keyboard shortcut. Again, like we do for all our other keyboard shortcuts!

yutong23 commented:
Solved ut!

misaugstad commented:
@yutong23 apologies, but we are actually going to put this PR on hold for now! At a meeting today, Jon and I were talking about how, with the addition of the 'missing crosswalk' tag, we would have to rethink how users validate Crosswalk labels. Right now we don't ask them to look at tags, so missing crosswalks labeled with the Crosswalk label would likely be rejected through the Validate interface.

This is making us rethink if we want to add this tag at all, so we are putting it on hold for now!

misaugstad commented:
Closing for now, since we've decided not to add this tag.
